ggu/main.c

```diff
@@ -53,40 +53,79 @@ process_single_policy(struct ggu_policy *policy, void *arg)
 {
 	const struct ggu_config *ggu_conf = arg;
 	struct gk_cmd_entry *entry;
-	/*
-	 * Obtain mailbox of that GK block,
-	 * and send the policy decision to the GK block.
-	 */
-	struct mailbox *mb =
-		get_responsible_gk_mailbox(&policy->flow, ggu_conf->gk);
-
-	if (mb == NULL)
-		return;
 
-	entry = mb_alloc_entry(mb);
-	if (entry == NULL)
-		return;
+	switch(policy->state) {
+	case GK_GRANTED: {
+		/*
+		 * Obtain mailbox of that GK block,
+		 * and send the policy decision to the GK block.
+		 */
+		struct mailbox *mb = get_responsible_gk_mailbox(
+			&policy->flow, ggu_conf->gk);
+		if (mb == NULL)
+			return;
 
-	entry->op = GGU_POLICY_ADD;
-	entry->u.ggu.state = policy->state;
-	rte_memcpy(&entry->u.ggu.flow, &policy->flow, sizeof(entry->u.ggu.flow));
+		entry = mb_alloc_entry(mb);
+		if (entry == NULL)
+			return;
 
-	switch (policy->state) {
-	case GK_GRANTED:
+		entry->op = GGU_POLICY_ADD;
+		entry->u.ggu.state = policy->state;
+		rte_memcpy(&entry->u.ggu.flow,
+			&policy->flow, sizeof(entry->u.ggu.flow));
 		entry->u.ggu.params.granted = policy->params.granted;
+		mb_send_entry(mb, entry);
 		break;
+	}
+
+	case GK_DECLINED: {
+		/*
+		 * Obtain mailbox of that GK block,
+		 * and send the policy decision to the GK block.
+		 */
+		struct mailbox *mb = get_responsible_gk_mailbox(
+			&policy->flow, ggu_conf->gk);
+		if (mb == NULL)
+			return;
+
+		entry = mb_alloc_entry(mb);
+		if (entry == NULL)
+			return;
 
-	case GK_DECLINED:
+		entry->op = GGU_POLICY_ADD;
+		entry->u.ggu.state = policy->state;
+		rte_memcpy(&entry->u.ggu.flow,
+			&policy->flow, sizeof(entry->u.ggu.flow));
 		entry->u.ggu.params.declined = policy->params.declined;
+		mb_send_entry(mb, entry);
 		break;
+	}
+
+	case GK_FLUSHED: {
+		/* This type is used for flushing policy decisions. */
+		int i;
+
+		for (i = 0; i < ggu_conf->gk->num_lcores; i++) {
+			entry = mb_alloc_entry(&ggu_conf->gk->instances[i].mb);
+			if (entry == NULL) {
+				RTE_LOG(WARNING, GATEKEEPER, "ggu: failed to allocate an entry from its mailbox\n");
```

Comment:
Review all uses of macro `RTE_LOG()` in this patch, not only the one above. Since we have merged a pull request that adds macros for each block. Use those macros instead of `RTE_LOG()`.